### PR TITLE
Save the libusb error if libusb_submit_transfer() fails

### PR DESCRIPTION
**Pull request by hughsie**
If the return code is LIBUSB_ERROR_NO_DEVICE we want to record that.

### CODE REVIEW

**Comment by hughsie** [comment]
This allows the TI dock to be emulated successfully! :)

File: gusb/gusb-device-event.c [hunk 1]
```diff
@@ -100,6 +102,10 @@ _g_usb_device_event_save(GUsbDeviceEvent *self, JsonBuilder *json_builder, GErro
 		json_builder_set_member_name(json_builder, "Status");
 		json_builder_add_int_value(json_builder, self->status);
 	}
+	if (self->rc != LIBUSB_SUCCESS) {
+		json_builder_set_member_name(json_builder, "Error");
```
Comment (nacho) [inline gusb/gusb-device-event.c:106]: do you want the error to be an object? that way it would be extendable in the future, to i.e have a description

Comment (hughsie) [reply gusb/gusb-device-event.c:106]: Maybe the name that's made it confusing? It's really just the raw error code from libusb, i.e. `-4` for `LIBUSB_ERROR_NO_DEVICE`

File: gusb/gusb-device.c [hunk 2]
```diff
@@ -2199,9 +2211,11 @@ g_usb_device_control_transfer_async(GUsbDevice *self,
 						event_id);
 			return;
 		}
-		if (g_usb_device_event_get_status(event) != LIBUSB_TRANSFER_COMPLETED) {
-			g_usb_device_libusb_status_to_gerror(g_usb_device_event_get_status(event),
-							     &error);
+		if (!g_usb_device_libusb_error_to_gerror(self,
```
Comment (nacho) [inline gusb/gusb-device.c:2214]: is g_usb_device_libusb_error_to_gerror setting the GError if it fails?

Comment (hughsie) [reply gusb/gusb-device.c:2214]: If rc == LIBUSB_SUCCESS then we return TRUE, i.e. don't set an error.

File: gusb/gusb-device-event.c (hunk 1 above)
Comment (nacho) [reply gusb/gusb-device-event.c:106]: I see... dunno then

File: gusb/gusb-device.c (hunk 2 above)
Comment (nacho) [reply gusb/gusb-device.c:2214]: ok, looks good then

**Review by nacho — APPROVED** [review]

File: gusb/gusb-device-event.c (hunk 1 above)
Comment (hughsie) [reply gusb/gusb-device-event.c:106]: Maybe it should be called `ErrorNumber`, `ErrorCode` or `Rc` or something?

Comment (hughsie) [reply gusb/gusb-device-event.c:106]: libusb just calls it `libusb_error`

**Review by fdanis-oss — APPROVED** [review]

**Pull request merged**